src/hotspot/share/runtime/nonJavaThread.cpp

```diff
@@ -235,6 +271,20 @@ int WatcherThread::sleep() const {
 void WatcherThread::run() {
   assert(this == watcher_thread(), "just checking");
 
+  while (true) {
+    // Just check for error reporting hangs until either VM is fully
+    // initialized or we are notified to stop running.
```

Comment:
Yes, I initially discarded that because sleep() would wait forever when there are no enrolled tasks yet, but not sure why I didn't just added a conditional for that case too inside sleep(). I removed the refactoring now and just added those conditionals.